### PR TITLE
feat: support html strings in title tooltip

### PR DESCRIPTION
**Pull request by arjunlalb**
## Description
Support html strings in title tooltip.

### Testing
UTs added.

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/components/src/table/header/table-header-cell-renderer.component.ts [hunk 1]
```diff
@@ -64,13 +76,18 @@ import { TableColumnConfigExtended } from '../table.service';
...
+          <div [innerHTML]="this.sanitizedHtmlForTooltip"></div>
+        </ng-template>
       </ng-template>
     </div>
   `
 })
 export class TableHeaderCellRendererComponent implements OnInit, OnChanges {
   public readonly SORT_ASC: TableSortDirection = TableSortDirection.Ascending;
   public readonly SORT_DESC: TableSortDirection = TableSortDirection.Descending;
+  private readonly HTML_TOOLTIP_PREFIX: string = 'html:';
```
Comment (aaron-steinfeld) [inline projects/components/src/table/header/table-header-cell-renderer.component.ts:90]: why the prefix? even a plain string is valid html so can we treat any tooltips that way?

File: projects/components/src/table/header/table-header-cell-renderer.component.ts [hunk 2]
```diff
@@ -102,9 +119,14 @@ export class TableHeaderCellRendererComponent implements OnInit, OnChanges {
 
   public isFilterable: boolean = false;
 
+  @ViewChild('htmlTooltip')
+  public htmlTooltipTemplate?: TemplateRef<unknown>;
+  public sanitizedHtmlForTooltip?: string | null;
```
Comment (aaron-steinfeld) [inline projects/components/src/table/header/table-header-cell-renderer.component.ts:124]: let's keep this as `?: string` (undefined or string) - we don't want to work with nulls, so wherever angular may return one, throw a `?? undefined` on.

File: projects/components/src/table/header/table-header-cell-renderer.component.ts (hunk 1 above)
Comment (arjunlalb) [reply projects/components/src/table/header/table-header-cell-renderer.component.ts:90]: I was being _clever_ with adding a differentiator to html strings. But you are right, its unnecessary. Removed.

File: projects/components/src/table/header/table-header-cell-renderer.component.ts (hunk 2 above)
Comment (arjunlalb) [reply projects/components/src/table/header/table-header-cell-renderer.component.ts:124]: Done

File: projects/components/src/table/header/table-header-cell-renderer.component.ts [hunk 3]
```diff
@@ -64,6 +76,10 @@ import { TableColumnConfigExtended } from '../table.service';
             </div>
           </ht-popover-content>
         </ht-popover>
+
+        <ng-template #htmlTooltip>
+          <div [innerHTML]="this.sanitizedHtmlForTooltip"></div>
```
Comment (aaron-steinfeld) [inline projects/components/src/table/header/table-header-cell-renderer.component.ts:81]: So I mentioned sanitization, but now that I think about it, angular should be doing that for us. Could you try removing the sanitizer and verifying that a tooltip is not executed (but still rendered?)

Ref: https://angular.io/guide/security#sanitization-example
Specifically:
> Angular recognizes the value as unsafe and automatically sanitizes it, which removes the onerror attribute but keeps safe content such as the <b> element.

File: projects/components/src/table/header/table-header-cell-renderer.component.ts [hunk 4]
```diff
@@ -144,6 +166,25 @@ export class TableHeaderCellRendererComponent implements OnInit, OnChanges {
...
+
+    return this.htmlTooltipTemplate;
+  }
+
+  public getSanitizedHtml(tooltipString: string | undefined): string | null {
+    if (tooltipString === undefined) {
+      return '';
+    }
+
+    return this.sanitizer.sanitize(SecurityContext.HTML, tooltipString);
```
Comment (aaron-steinfeld) [inline projects/components/src/table/header/table-header-cell-renderer.component.ts:185]: if we keep the sanitizer, we should put the `?? undefined` here - our methods should never return null (I thought we had a lint rule on that)

File: projects/components/src/table/header/table-header-cell-renderer.component.ts (hunk 3 above)
Comment (arjunlalb) [reply projects/components/src/table/header/table-header-cell-renderer.component.ts:81]: aah. cool. it does do the work for us.

Comment (arjunlalb) [reply projects/components/src/table/header/table-header-cell-renderer.component.ts]: Updated.

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**